src/code_80086E70.c

```diff
@@ -580,226 +580,344 @@ void func_800887C0(s32 objectIndex) {
 }
 
-s32 func_80088AC0(s32 objectIndex, Player *player, f32 arg2, f32 arg3) {
-    f32 temp_f0;
-    f32 temp_f12;
-    f32 temp_f2;
-    s32 var_v1;
-
-    var_v1 = 0;
-    temp_f0 = gObjectList[objectIndex].pos[0] - player->pos[0];
-    temp_f2 = gObjectList[objectIndex].pos[2] - player->pos[2];
-    temp_f12 = (temp_f0 * temp_f0) + (temp_f2 * temp_f2);
-    if (((arg2 * arg2) <= temp_f12) && (temp_f12 <= (arg3 * arg3))) {
-        var_v1 = 1;
+/**
+ * @brief Returns if the object is between the distance_min and distance_max to the player.
+ * 
+ * @param objectIndex 
+ * @param player 
+ * @param distance_min 
+ * @param distance_max 
+ * @return true 
+ * @return false 
+ */
+bool are_between_distance_player_2d(s32 objectIndex, Player *player, f32 distance_min, f32 distance_max) {
+    f32 x;
+    f32 distance;
+    f32 y;
+    bool var_v1;
+
+    var_v1 = FALSE;
+    x = gObjectList[objectIndex].pos[0] - player->pos[0];
+    y = gObjectList[objectIndex].pos[2] - player->pos[2];
+    distance = (x * x) + (y * y);
+    if (((distance_min * distance_min) <= distance) && (distance <= (distance_max * distance_max))) {
+        var_v1 = TRUE;
     }
     return var_v1;
 }
 
-s32 func_80088B40(s32 objectIndex, Player *player, f32 arg2) {
-    f32 temp_f0;
-    f32 temp_f12;
-    f32 temp_f2;
-    s32 var_v1;
-
-    var_v1 = 0;
-    temp_f0  = gObjectList[objectIndex].pos[0] - player->pos[0];
-    temp_f2  = gObjectList[objectIndex].pos[1] - player->pos[1];
-    temp_f12 = gObjectList[objectIndex].pos[2] - player->pos[2];
-    if (((temp_f0 * temp_f0) + (temp_f2 * temp_f2) + (temp_f12 * temp_f12)) <= (arg2 * arg2)) {
-        var_v1 = 1;
+/**
+ * @brief Returns if the object is closer than the distance to the player.
+ * 
+ * @param objectIndex 
+ * @param player 
+ * @param distance 
+ * @return true 
+ * @return false 
+ */
+bool is_closer_of_player_3d(s32 objectIndex, Player *player, f32 distance) {
+    f32 x;
+    f32 z;
+    f32 y;
+    bool var_v1;
+
+    var_v1 = FALSE;
+    x  = gObjectList[objectIndex].pos[0] - player->pos[0];
+    y  = gObjectList[objectIndex].pos[1] - player->pos[1];
+    z = gObjectList[objectIndex].pos[2] - player->pos[2];
+    if (((x * x) + (y * y) + (z * z)) <= (distance * distance)) {
+        var_v1 = TRUE;
     }
     return var_v1;
 }
 
-s32 func_80088BB8(s32 objectIndex, Camera *camera, f32 arg2) {
-    f32 temp_f0;
-    f32 temp_f2;
-    s32 var_v1;
-
-    var_v1 = 0;
-    temp_f0 = gObjectList[objectIndex].pos[0] - camera->pos[0];
-    temp_f2 = gObjectList[objectIndex].pos[2] - camera->pos[2];
-    if (((temp_f0 * temp_f0) + (temp_f2 * temp_f2)) <= (arg2 * arg2)) {
-        var_v1 = 1;
+/**
+ * @brief Returns if the object is between the distance_min and distance_max to the camera.
+ * 
+ * @param objectIndex 
+ * @param camera 
+ * @param distance 
+ * @return true 
+ * @return false 
+ */
+bool is_closer_of_camera_2d(s32 objectIndex, Camera *camera, f32 distance) {
+    f32 x;
+    f32 y;
```

Comment:
x and z
is_object_within_horizontal_distance_to_camera
or
is_within_horizontal_distance_to_camera
(or similar)